### PR TITLE
feat(runtime-core-tests) Introduce the new `wasmer-runtime-core-tests` crate

### PR DESCRIPTION
**Pull request by Hywan**
Extracted from https://github.com/wasmerio/wasmer/pull/882.

This non-publishable new crate contains a test suite for the
`wasmer-runtime-core` crate. So far, the test suite is rather small,
but it aims to be extended in a close future.

### CODE REVIEW

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors cancel

**Comment by Hywan** [comment]
bors try

File: lib/runtime-core-tests/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "wasmer-runtime-core-tests"
+version = "0.8.0"
```
Comment (MarkMcCaskey) [inline lib/runtime-core-tests/Cargo.toml]: This should be 0.9.0 now!

**Pull request merged**